controller/controller.go

```diff
@@ -203,18 +200,26 @@ func WebStoreUpdateExtension(w http.ResponseWriter, r *http.Request) {
 
 	w.WriteHeader(http.StatusOK)
 
-	if IsJSONRequest(r.Header.Get("Content-Type")) {
-		w.Header().Set("content-type", "application/json")
-		data, err = json.Marshal(&webStoreResponse)
-	} else {
-		w.Header().Set("content-type", "application/xml")
-		data, err = xml.Marshal(&webStoreResponse)
+	// Always use protocol 3.1 for responses
+	protocolHandler, err := ProtocolFactory.CreateProtocol("3.1")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error creating protocol handler: %v", err), http.StatusInternalServerError)
+		return
 	}
 
+	data, err := protocolHandler.FormatWebStoreResponse(webStoreResponse, contentType)
 	if err != nil {
-		http.Error(w, fmt.Sprintf("Error in marshal %v", err), http.StatusInternalServerError)
+		http.Error(w, fmt.Sprintf("Error formatting response: %v", err), http.StatusInternalServerError)
 		return
 	}
+
+	// Set content type
+	if protocol.IsJSONRequest(contentType) {
+		w.Header().Set("content-type", "application/json")
```

Comment:
I think I shared elsewhere - but if `contentType` is guaranteed to be `application/json` or `application/xml`, you could simplify this:

```
w.Header().Set("content-type", contentType)
```

Reply:
whoops, missed this comment https://github.com/brave/go-update/pull/255#discussion_r2036298314

Closing; it's fine